watchmal/engine/engine_classifier.py

```diff
@@ -143,7 +143,7 @@ def forward(self, train=True):
         """
         with torch.set_grad_enabled(train):
             # Move the data and the labels to the GPU (if using CPU this has no effect)
-            data = self.data.to(self.device)
+            data = self.data.to(self.device, dtype=torch.float)
```

Comment:
This shouldn't be necessary. We put quite a lot of work into minimising type conversions here to speed up training (it's better to do it in the data loader where there's multiple workers if needed at all). Could you check compared to the mpmt dataset to see why it's needed for you, so it can be removed from the engine code?

Reply:
I see that there was a pytorch upgrade among the upgrades to the singularity container. It seems that this new error was resolved in the cnn_mpmt class by initializing the input numpy array as dtype=float32, which I had not noticed. I implemented this change and removed changes to the engine.